diff --git a/src/Controller/NewS3fsImageStyleDownloadController.php b/src/Controller/NewS3fsImageStyleDownloadController.php
index fd520a2..066c1c5 100644
--- a/src/Controller/NewS3fsImageStyleDownloadController.php
+++ b/src/Controller/NewS3fsImageStyleDownloadController.php
@@ -261,7 +261,7 @@ final class NewS3fsImageStyleDownloadController implements ContainerInjectionInt
 
       $responseCacheTags = $image_style->getCacheTags();
 
-      // Try to get a managed file and flush the cache.
+      // Flush the cache of every managed file that has this URI.
       $storage = $this->entityTypeManager
         ->getStorage('file');
       $result = $storage
@@ -280,7 +280,6 @@ final class NewS3fsImageStyleDownloadController implements ContainerInjectionInt
           if ($file->getFileUri() === $image_uri) {
             Cache::invalidateTags($file->getCacheTags());
             $responseCacheTags = Cache::mergeTags($responseCacheTags, $file->getCacheTags());
-            break;
           }
         }
       }
diff --git a/src/Controller/S3fsImageStyleDownloadController.php b/src/Controller/S3fsImageStyleDownloadController.php
index 80a55c7..51f2ea5 100644
--- a/src/Controller/S3fsImageStyleDownloadController.php
+++ b/src/Controller/S3fsImageStyleDownloadController.php
@@ -155,7 +155,7 @@ class S3fsImageStyleDownloadController extends ImageStyleDownloadController {
 
       $responseCacheTags = $image_style->getCacheTags();
 
-      // Try to get a managed file and flush the cache.
+      // Flush the cache of every managed file that has this URI.
       $storage = $this->entityTypeManager()
         ->getStorage('file');
       $result = $storage
@@ -174,7 +174,6 @@ class S3fsImageStyleDownloadController extends ImageStyleDownloadController {
           if ($file->getFileUri() === $image_uri) {
             Cache::invalidateTags($file->getCacheTags());
             $responseCacheTags = Cache::mergeTags($responseCacheTags, $file->getCacheTags());
-            break;
           }
         }
       }
